.github/libs/GitUtils.js

```diff
@@ -36,10 +37,8 @@ function fetchTagIfNeeded(tag) {
  * @returns {Promise<Array<Object<{commit: String, subject: String, authorName: String}>>>}
  */
 function getCommitHistoryAsJSON(fromTag, toTag) {
-    // fetchTagIfNeeded(fromTag);
-    // fetchTagIfNeeded(toTag);
-    // Note: this is a temporary measure until we can figure out a faster way to fetch only what's needed
-    execSync('git fetch --all --tags');
+    fetchTag(fromTag);
+    fetchTag(toTag);
```

Comment:
Good point

Reply:
Yes @luacmartins, that's correct. Great observation.

However, if you do: `git diff 1.1.0...1.2.0`, then you're asking git "give me all the commits reachable by `1.2.0` but not by `1.1.0`". So as you observed:

- If you checkout `1.1.0` then do a full `git log`, all the history from `1.0.0` to `1.1.0` is reachable.
- If you checkout `1.2.0` then do a full `git log`, all the history from `1.1.0` to `1.2.0` is reachable.

So if you carry that out, `git diff 1.1.0...1.2.0` will give you what you expect – all the history from `1.1.0` to `1.2.0`.

---

If you're still with me, then I'm sure you have more questions.

**Question:** Since a full `git log` from `1.2.0` appears to give us the same thing, then why don't we just do that, and skip fetching the history from `1.0.0` to `1.1.0`?

**Answer:**

1. Start at `1.0.0-0`
1. PR 1 merged
1. Version bumped to `1.0.0-1`
1. PR 2 merged
1. Version bumped to `1.0.0-2`
1. Now we need to know two separate diffs:
    1. `1.0.0-0...1.0.0-2` to tell us the full list of PRs to include in the deploy checklist
    1. `1.0.0-1...1.0.0-2` to tell us which PR was just deployed and needs a deploy comment.
1. If we do a full `git log` from `1.0.0-2`, it will give us both PRs, `1` and `2`. So we can't easily tell which PR was just deployed. ❌ 

---

**Question:** What if we just do a `git fetch origin tag ${fromTag} --depth=1`, so that we have the full history of the `toTag` and a ref to the `fromTag`?

**Answer:**

1. Start at `1.0.0-0`
1. PR 1 merged
1. Version bumped to `1.0.0-1`
1. PR 2 merged. Checklist is locked so version not bumped.
1. PR 3 merged (and cherry-picked)
1. Version bumped to `1.0.0-2`
1. PR 4 created to cherry-pick commits to staging (note: this PR doesn't happen on every CP, but is unavoidable in the event of conflicts)
1. Checklist closed, production deploy happens, version bumped to `1.0.1-0`
1. Now we do `git log 1.0.0-2...1.0.1-0`. What do we get?
    1. `1.0.0-2` is a reference to a single commit, just the merge commit of the CP PR from `main` -> `staging`
    1. `1.0.1-0` has all the history from `1.0.0-0` to `1.0.1-0`
    1. So `git log 1.0.0-2...1.0.1-0` gives us all the history between `1.0.0-0` and `1.0.1-0` because nothing is reachable from `1.0.0-2`. ❌ 
    
---

**Question:** What if we do `git fetch origin tag ${fromTag} --depth=1` and `git fetch origin tag ${toTag} --shallow-exclude ${fromTag}`?

**Answer:**

This fails [here](https://github.com/Expensify/App/blob/389412b6bdf892e5d71ef05caf0295a416b1c9e4/tests/unit/CIGitLogicTest.sh#L240), I'm still a little unclear why though... working on finding a suitable explanation for that answer...

Reply:
FWIW though, the history between the current checklist and the previous one will have a reasonable small cap, where it should only take at most a few seconds to fetch, as opposed to about 1-2 minutes (and this number will only grow) for a full fetch of all history and all branches that we get from actions/checkout w/ `fetch-depth: 0`

Reply:
Thanks for clarifying and all the additional info! I think this makes sense and should work well.